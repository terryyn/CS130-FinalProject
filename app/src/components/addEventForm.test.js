import React from 'react';
import StyleForm from './addeventForm.js';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import {shallow,configure } from 'enzyme';


configure({adapter: new Adapter()});

describe('add event form', () =>{
  let  component;
  beforeEach (() => {
    component = shallow(<StyleForm/>).dive();
  });

  test('snapshot renders', () => {
    const component = renderer.create(<StyleForm/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('handle start date change', () => {
    const event = {
      target: {value: '2019-11-15'}
    };
    
    component.find('#startdate-input').simulate('change', event);
    expect(component.state('startdate')).toBe('2019-11-15');
  });

  test('handle start time change', () => {
    const event = {
      target: {value: '14:15'}
    };
    
    component.find('#starttime-input').simulate('change', event);
    expect(component.state('starttime')).toBe('14:15');
  });

  test('handle name change', () => {
    const event = {
      target: {value: 'class'}
    };
    
    component.find('#name-input').simulate('change', event);
    expect(component.state('name')).toBe('class');
  });

  test('handle type change', () => {
    const event = {
      target: {value: 2}
    };
    
    component.find('#type-input').simulate('change', event);
    expect(component.state('type')).toBe(2);
  });
});