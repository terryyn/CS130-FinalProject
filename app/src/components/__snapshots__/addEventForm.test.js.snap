// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add event form snapshot renders 1`] = `
<div
  className="addevent"
>
  <h1
    className="addevent-title"
  >
    Add an event
  </h1>
  <form
    className="addevent-form"
  >
    <div
      className="addevent-field"
      id="start"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
        id="start-label"
      >
        Start
      </label>
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input AddEventForm-textField-1"
            disabled={false}
            id="startdate-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="date"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-65 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-66"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        id="starttime"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input AddEventForm-textField-1"
              disabled={false}
              id="starttime-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="time"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-65 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-66"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="addevent-field"
      id="end"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
        id="end-label"
      >
        End
      </label>
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input AddEventForm-textField-1"
            disabled={false}
            id="enddate-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="date"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-65 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-66"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        id="endtime"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input AddEventForm-textField-1"
              disabled={false}
              id="endtime-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="time"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-65 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-66"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="addevent-field"
      id="name"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
        id="name-label"
      >
        Name
      </label>
      <div
        className="MuiFormControl-root MuiTextField-root"
        title="name"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input AddEventForm-textField-1"
            disabled={false}
            id="name-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-65 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-66"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="addevent-field"
      id="location"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
        id="location-label"
      >
        Location
      </label>
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input AddEventForm-textField-1"
            disabled={false}
            id="location-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-65 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-66"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="addevent-field"
      id="type"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
        id="type-label"
      >
        Type
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline"
        label="Type"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby=" type-input"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
          id="type-input"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          onAnimationStart={[Function]}
          type="hidden"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      className="addevent-field"
      id="description"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
        id="description-label"
      >
        Description
      </label>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
          onClick={[Function]}
        >
          <textarea
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            disabled={false}
            id="description-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            rows="2"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-65 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-66"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      id="submit"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        id="submit-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "margin": "10px 10px 10px 0",
          }
        }
        tabIndex={0}
        title="Submit"
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Create Event
        </span>
      </button>
    </div>
  </form>
</div>
`;
